static/panes/compiler.js

```diff
@@ -321,7 +350,72 @@ Compiler.prototype.resize = function () {
     });
 };
 
+// Returns a label name if it can be found in the given position, otherwise
+// returns null.
+Compiler.prototype.getLabelAtPosition = function (position) {
+    var line = this.outputEditor.getModel().getLineContent(position.lineNumber);
+    var match = line.match(this.labelUsageRegex);
+    if (match) {
+        return match[2].trim();
+    }
+
+    return null;
+};
+
+// Jumps to a label definition related to a label which was found in the
+// given position and highlights the given range. If no label can be found in
+// the given positon it do nothing.
+Compiler.prototype.jumpToLabel = function (position) {
+    var label = this.getLabelAtPosition(position);
+
+    if (!label) {
+        return;
+    }
+
+    var range = this.labels[label];
```

Comment:
similarly in a server-side world we'd populate `this.labels` in `setAssembly` by looking for a `labelDefinition` or similar tag. We can also add "decorations" in `setAssembly` to make the labels look like hyperlinks.